src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/entity/EnvelopeRepositoryTest.java

```diff
@@ -126,6 +128,57 @@ public void should_get_1_incomplete_envelope_from_yesterday() {
         assertThat(repo.getIncompleteEnvelopesCountBefore(now().toLocalDate())).isEqualTo(1);
     }
 
+    @Test
+    public void should_find_complete_envelopes_by_container() {
+        // given
+        final String container1 = "container1";
+        dbHas(
+            envelope("X", Status.COMPLETED, container1),
+            envelope("Y", Status.PROCESSED, "container2"),
+            envelope("X", Status.COMPLETED, container1),
+            envelope("X", Status.PROCESSED, container1),
+            envelope("Z", Status.COMPLETED, "container3")
+        );
+
+        // when
+        final List<Envelope> result = repo.findByContainerAndStatus(container1, Status.COMPLETED);
+
+        // then
+        assertThat(result)
+            .hasSize(2)
+            .extracting(this::getTupleFromEnvelope)
```

Comment:
could be  oneliner anyway: `envelope -> tuple(envelope.status, envelope.container)`

Reply:
fixed